docs/docs/user-guide/training-data-format.mdx

```diff
@@ -0,0 +1,659 @@
+mapped to the **value** `credit`.
+
+:::note
+Synonym mapping only happens **after** entities have been extracted.
+That means that you need to provide examples of the variations 
+on a synonym in your training examples so
+that Rasa Open Source can learn to pick them up.
+:::
+
+### Regular Expression Features
+
+Regular expressions can be used in two different ways:
+
+1. They can be used to support intent classification and entity
+extraction when using the [RegexFeaturizer](../nlu/components#regexfeaturizer) component in the pipeline.
+
+2. They can be used to extract entities from a user message using the rule-based
+[`RegexEntityExtractor`](../nlu/components#regexentityextractor) component in the pipeline.
+
+For example, if your entity has a deterministic structure (like a
+zipcode or an email address), you can use a regular expression to ease
+detection of that entity (using the `RegexFeaturizer`) or to
+extract the entities from the user message directly (using the
+`RegexEntityExtractor`). For a bank account number that is 10-12 digits long,
+it might look like this:
+
+```yaml
+- regex: account_number
+  examples: |
+  - \d{10:12}
+```
+
+If you are using regular expressions to directly extract entities using
+the `RegexEntityExtractor`, the name of the regular expression should
+match the name of the entity you want to extract.
+
+If you are using the regular expressions for the `RegexFeaturizer` the
+name of the regular expression does not matter. If does not define the
+entity nor the intent, it is a human readable description for you
+to remember what this regex is used for and is the title of the
+corresponding pattern feature.
+
+If you want to use the RegexFeaturizer you can also use the regex
+features to improve the intent classification performance, for example,
+by defining a greet clause:
+
+```yaml
+- regex: greet
+  examples: |
+  - hey[^\\s]*
+```
+
+Try to create your regular expressions in a way that they match as few
+words as possible. E.g. using `hey[^\\s]*` instead of `hey.*`, as the
+later one might match the whole message whereas the first one only
+matches a single word.
+
+When using the `RegexFeaturizer`, the regex features for entity extraction
+are currently only supported by the `CRFEntityExtractor` and the
+`DIETClassifier` component! Hence, other entity extractors, like
+`MitieEntityExtractor` or `SpacyEntityExtractor` won't use the generated
+features and their presence will not improve entity recognition for
+these extractors. Currently, all intent classifiers make use of
+available regex features.
+
+:::note
+
+Regex features only define entities when used in combination with the `RegexEntityExtractor`. Otherwise they  
+don't define entities nor intents! They only provide patterns to
+help the classifier recognize entities and related intents. Hence, you
+still need to provide intent & entity examples as part of your
+training data!
+:::
```

Comment:
```suggestion
:::note

When using the `RegexFeaturizer`, a regular expression only provides a feature
that helps the model learn an association between intents/entities and inputs
that fit the regular expression. In order for the model to learn this association,
you must provide example inputs that include that regular expression! 
:::
```
Do you think this makes more sense? I'm trying to rephrase it as what to do when using features (versus when not using direct extraction)